scripts/build/utils/dts-process.ts

```diff
@@ -0,0 +1,67 @@
+import { writeFile } from 'node:fs/promises';
+import { dirname, join, sep } from 'node:path';
+
+import { rollup } from 'rollup';
+import { dts } from 'rollup-plugin-dts';
+import { JsxEmit, ScriptTarget } from 'typescript';
+
+import { getExternal } from './entry-utils';
+
+async function run() {
+  const [entryPoint] = process.argv.slice(2);
+
+  if (!entryPoint) {
+    throw new Error(
+      'No entry point provided. Usage: jiti scripts/build/utils/dts-process.ts <entryPoint>'
+    );
+  }
+  const { typesExternal: external } = await getExternal(process.cwd());
+
+  const dir = dirname(entryPoint).replace('src', 'dist');
+  const outputFile = entryPoint.replace('src', 'dist').replace('.ts', '.d.ts');
+  const out = await rollup({
+    input: entryPoint,
+    external: (id) => {
+      return external.some(
+        (dep) => id.startsWith(dep) || id.includes(`${sep}node_modules${sep}${dep}${sep}`)
+      );
+    },
+    output: { file: outputFile, format: 'es' },
+    plugins: [
+      dts({
+        respectExternal: true,
+        tsconfig: join(process.cwd(), 'tsconfig.json'),
+        compilerOptions: {
+          esModuleInterop: true,
+          baseUrl: '.',
+          jsx: JsxEmit.React,
```

Comment:
Do you know if this always true? Even for building assets related to Preact, for example?

Reply:
This is copied from the core build script, so if it was a problem I'd expect we'd have heard about it for now. It doesn't look like we're building any Preact-specific JSX where we would have wanted to change this anyway.